extensions/amp-story/1.0/amp-story-store-service.js

```diff
@@ -222,6 +237,39 @@ const stateComparisonFunctions = {
     (old, curr) => old.element !== curr.element || old.state !== curr.state,
   [StateProperty.NAVIGATION_PATH]: (old, curr) => old.length !== curr.length,
   [StateProperty.PAGE_IDS]: (old, curr) => old.length !== curr.length,
+  [StateProperty.INTERACTIVE_RESULTS_STATE]: (old, curr) => {
+    return old.finished != curr.finished;
+  },
+};
+
+/**
+ * Gets the map of quizzes responded and sets the correct values on the template.
+ * @param {!Object} data
+ * @return {!InteractiveResultsDef}
+ */
+const processInteractiveResults = (data) => {
+  let completed = 0;
+  let totalCount = 0;
+  const categories = {null: -1};
+  Object.values(data).forEach((interactiveConfig) => {
+    totalCount += 1;
+    if (interactiveConfig['option'] != null) {
+      completed += 1;
+      const currCategory = interactiveConfig['option']['category'];
+      if (currCategory != undefined) {
+        categories[currCategory] = categories[currCategory]
+          ? categories[currCategory] + 1
+          : 1;
+      }
+    }
+  });
+  const result = {
+    'finished': completed == totalCount,
+    'category': Object.keys(categories).reduce((prev, curr) =>
+      categories[prev] > categories[curr] ? prev : curr
+    ),
+  };
```

Comment:
Counts the category fields and selects the one that has the most votes across all the interactive components